<h1>Activity 1 - PALMER PENGUINS DATA ANALYTICS PROJECT </h1>
<hr>
<h3> Analyst : Louie John Lusanta </h3>

In [1]:
#import libraries
import numpy as np
import pandas as pd

In [2]:
penguins = pd.read_csv('C:\\Users\\DESIREE\\Downloads\\datasets\\penguins.csv')

In [3]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

In [4]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [5]:
penguins.describe(). T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

<h2>Checking Null Values</h2>

In [6]:
penguins.isnull()

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0      False   False           False          False              False   
1      False   False           False          False              False   
2      False   False           False          False              False   
3      False   False            True           True               True   
4      False   False           False          False              False   
..       ...     ...             ...            ...                ...   
339    False   False           False          False              False   
340    False   False           False          False              False   
341    False   False           False          False              False   
342    False   False           False          False              False   
343    False   False           False          False              False   

     body_mass_g    sex   year  
0          False  False  False  
1          False  False  False  
2          False  False  False  
3           True   True  False  
4          False  False  False  
..           ...    ...    ...  
339        False  False  False  
340        False  False  False  
341        False  False  False  
342        False  False  False  
343        False  False  False  

[344 rows x 8 columns]

In [7]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h2>Filtering A Column</h2>

In [8]:
penguins['species']

0         Adelie
1         Adelie
2         Adelie
3         Adelie
4         Adelie
         ...    
339    Chinstrap
340    Chinstrap
341    Chinstrap
342    Chinstrap
343    Chinstrap
Name: species, Length: 344, dtype: object

In [9]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [10]:
penguins['species'] == 'Chinstrap'

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: species, Length: 344, dtype: bool

In [11]:
penguins[penguins['species'] == 'Chinstrap']

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [12]:
adelie_df = penguins[penguins['species'] == 'Chinstrap']
adelie_df

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [13]:
gentoo_df = penguins[penguins['species'] == 'Gentoo']
gentoo_df

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

In [14]:
penguins['species'] == 'Chinstrap'

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: species, Length: 344, dtype: bool

In [15]:
 (penguins['body_mass_g'] >= 4000)

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340    False
341    False
342     True
343    False
Name: body_mass_g, Length: 344, dtype: bool

In [16]:
(penguins['species'] == 'Chinstrap') & (penguins['body_mass_g'] >= 4000)

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340    False
341    False
342     True
343    False
Length: 344, dtype: bool

In [17]:
penguins[(penguins['species'] == 'Chinstrap') & (penguins['body_mass_g'] >= 4000)]

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
284  Chinstrap  Dream            46.0           18.9              195.0   
289  Chinstrap  Dream            52.0           18.1              201.0   
291  Chinstrap  Dream            50.5           19.6              201.0   
295  Chinstrap  Dream            49.2           18.2              195.0   
301  Chinstrap  Dream            52.0           19.0              197.0   
305  Chinstrap  Dream            52.8           20.0              205.0   
307  Chinstrap  Dream            54.2           20.8              201.0   
309  Chinstrap  Dream            51.0           18.8              203.0   
313  Chinstrap  Dream            52.0           20.7              210.0   
315  Chinstrap  Dream            53.5           19.9              205.0   
321  Chinstrap  Dream            50.8           18.5              201.0   
323  Chinstrap  Dream            49.0           19.6              212.0   
329  Chinstrap  Dream            50.7           19.7              203.0   
333  Chinstrap  Dream            49.3           19.9              203.0   
339  Chinstrap  Dream            55.8           19.8              207.0   
342  Chinstrap  Dream            50.8           19.0              210.0   

     body_mass_g     sex  year  
284       4150.0  female  2007  
289       4050.0    male  2007  
291       4050.0    male  2007  
295       4400.0    male  2007  
301       4150.0    male  2007  
305       4550.0    male  2008  
307       4300.0    male  2008  
309       4100.0    male  2008  
313       4800.0    male  2008  
315       4500.0    male  2008  
321       4450.0    male  2009  
323       4300.0    male  2009  
329       4050.0    male  2009  
333       4050.0    male  2009  
339       4000.0    male  2009  
342       4100.0    male  2009

In [18]:
chinstrap_heavy = penguins[(penguins['species'] == 'Chinstrap') & (penguins['body_mass_g'] >= 4000)]
chinstrap_heavy

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
284  Chinstrap  Dream            46.0           18.9              195.0   
289  Chinstrap  Dream            52.0           18.1              201.0   
291  Chinstrap  Dream            50.5           19.6              201.0   
295  Chinstrap  Dream            49.2           18.2              195.0   
301  Chinstrap  Dream            52.0           19.0              197.0   
305  Chinstrap  Dream            52.8           20.0              205.0   
307  Chinstrap  Dream            54.2           20.8              201.0   
309  Chinstrap  Dream            51.0           18.8              203.0   
313  Chinstrap  Dream            52.0           20.7              210.0   
315  Chinstrap  Dream            53.5           19.9              205.0   
321  Chinstrap  Dream            50.8           18.5              201.0   
323  Chinstrap  Dream            49.0           19.6              212.0   
329  Chinstrap  Dream            50.7           19.7              203.0   
333  Chinstrap  Dream            49.3           19.9              203.0   
339  Chinstrap  Dream            55.8           19.8              207.0   
342  Chinstrap  Dream            50.8           19.0              210.0   

     body_mass_g     sex  year  
284       4150.0  female  2007  
289       4050.0    male  2007  
291       4050.0    male  2007  
295       4400.0    male  2007  
301       4150.0    male  2007  
305       4550.0    male  2008  
307       4300.0    male  2008  
309       4100.0    male  2008  
313       4800.0    male  2008  
315       4500.0    male  2008  
321       4450.0    male  2009  
323       4300.0    male  2009  
329       4050.0    male  2009  
333       4050.0    male  2009  
339       4000.0    male  2009  
342       4100.0    male  2009

<h2> Filling UP Null Values <h2>

In [19]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [20]:
penguins['bill_length_mm'] 

0      39.1
1      39.5
2      40.3
3       NaN
4      36.7
       ... 
339    55.8
340    43.5
341    49.6
342    50.8
343    50.2
Name: bill_length_mm, Length: 344, dtype: float64

In [21]:
penguins['bill_length_mm'] . mean()

np.float64(43.9219298245614)

In [22]:
penguins['bill_length_mm'] = penguins['bill_length_mm'] .fillna(penguins['bill_length_mm'].mean())

In [23]:
penguins['bill_depth_mm'] = penguins['bill_depth_mm'] .fillna(penguins['bill_depth_mm'].mean())


In [24]:
penguins['flipper_length_mm'] = penguins['flipper_length_mm'] .fillna(penguins['flipper_length_mm'].mean())


In [25]:
penguins['body_mass_g'] = penguins['body_mass_g'] .fillna(penguins['body_mass_g'].mean())


In [26]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           0
sex                  11
year                  0
dtype: int64

In [27]:
penguins['sex'] = penguins['sex'] .fillna(penguins['sex'] . mode()[0])

In [28]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

<h2> CHANGING THE DATA TYPE <h2>

In [29]:
penguins['species'] = penguins['species'].astype('category')
penguins['island'] = penguins['island'].astype('category')
penguins['sex'] = penguins['sex'].astype('category')
penguins['year'] = penguins['year'].astype('category')





In [30]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
dtypes: category(4), float64(4)
memory usage: 12.7 KB


In [31]:
penguins['body_mass_g'].mean()

np.float64(4201.754385964912)

In [32]:
penguins['species'] == 'Gentoo'

0      False
1      False
2      False
3      False
4      False
       ...  
339    False
340    False
341    False
342    False
343    False
Name: species, Length: 344, dtype: bool

<h4>How to Create a Column</h4>

In [33]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
dtypes: category(4), float64(4)
memory usage: 12.7 KB


In [34]:
penguins.columns

Index(['species', 'island', 'bill_length_mm', 'bill_depth_mm',
       'flipper_length_mm', 'body_mass_g', 'sex', 'year'],
      dtype='object')

In [35]:
penguins['mass_kg'] = np.round(penguins['body_mass_g'] / 1000 , 1)

In [36]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
1       Adelie  Torgersen        39.50000       17.40000         186.000000   
2       Adelie  Torgersen        40.30000       18.00000         195.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
4       Adelie  Torgersen        36.70000       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000       19.80000         207.000000   
340  Chinstrap      Dream        43.50000       18.10000         202.000000   
341  Chinstrap      Dream        49.60000       18.20000         193.000000   
342  Chinstrap      Dream        50.80000       19.00000         210.000000   
343  Chinstrap      Dream        50.20000       18.70000         198.000000   

     body_mass_g     sex  year  mass_kg  
0    3750.000000    male  2007      3.8  
1    3800.000000  female  2007      3.8  
2    3250.000000  female  2007      3.2  
3    4201.754386    male  2007      4.2  
4    3450.000000  female  2007      3.4  
..           ...     ...   ...      ...  
339  4000.000000    male  2009      4.0  
340  3400.000000  female  2009      3.4  
341  3775.000000    male  2009      3.8  
342  4100.000000    male  2009      4.1  
343  3775.000000  female  2009      3.8  

[344 rows x 9 columns]

In [37]:
penguins['mass_kg'].max()

np.float64(6.3)

penguins[penguins['mass_kg'] == 6.3]

<h2> EXPLANATORY DATA ANALYTICS <h2>

<h2>Question 1: What is the lightest body mass of female adelie in the year 2008? </h2

In [59]:
female_adelie_2008 = penguins[(penguins["species"] == "Adelie") & (penguins["sex"] == "female") & (penguins["year"] == 2008)]
lightest_mass = female_adelie_2008["body_mass_g"].min()


In [60]:
female_adelie_2008

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
50  Adelie     Biscoe            39.6           17.7              186.0   
52  Adelie     Biscoe            35.0           17.9              190.0   
54  Adelie     Biscoe            34.5           18.1              187.0   
56  Adelie     Biscoe            39.0           17.5              186.0   
58  Adelie     Biscoe            36.5           16.6              181.0   
60  Adelie     Biscoe            35.7           16.9              185.0   
62  Adelie     Biscoe            37.6           17.0              185.0   
64  Adelie     Biscoe            36.4           17.1              184.0   
66  Adelie     Biscoe            35.5           16.2              195.0   
68  Adelie  Torgersen            35.9           16.6              190.0   
70  Adelie  Torgersen            33.5           19.0              190.0   
72  Adelie  Torgersen            39.6           17.2              196.0   
74  Adelie  Torgersen            35.5           17.5              190.0   
76  Adelie  Torgersen            40.9           16.8              191.0   
78  Adelie  Torgersen            36.2           16.1              187.0   
80  Adelie  Torgersen            34.6           17.2              189.0   
82  Adelie  Torgersen            36.7           18.8              187.0   
84  Adelie      Dream            37.3           17.8              191.0   
87  Adelie      Dream            36.9           18.6              189.0   
89  Adelie      Dream            38.9           18.8              190.0   
90  Adelie      Dream            35.7           18.0              202.0   
92  Adelie      Dream            34.0           17.1              185.0   
94  Adelie      Dream            36.2           17.3              187.0   
96  Adelie      Dream            38.1           18.6              190.0   
98  Adelie      Dream            33.1           16.1              178.0   

    body_mass_g     sex  year  mass_kg  
50       3500.0  female  2008      3.5  
52       3450.0  female  2008      3.4  
54       2900.0  female  2008      2.9  
56       3550.0  female  2008      3.6  
58       2850.0  female  2008      2.8  
60       3150.0  female  2008      3.2  
62       3600.0  female  2008      3.6  
64       2850.0  female  2008      2.8  
66       3350.0  female  2008      3.4  
68       3050.0  female  2008      3.0  
70       3600.0  female  2008      3.6  
72       3550.0  female  2008      3.6  
74       3700.0  female  2008      3.7  
76       3700.0  female  2008      3.7  
78       3550.0  female  2008      3.6  
80       3200.0  female  2008      3.2  
82       3800.0  female  2008      3.8  
84       3350.0  female  2008      3.4  
87       3500.0  female  2008      3.5  
89       3600.0  female  2008      3.6  
90       3550.0  female  2008      3.6  
92       3400.0  female  2008      3.4  
94       3300.0  female  2008      3.3  
96       3700.0  female  2008      3.7  
98       2900.0  female  2008      2.9

In [61]:
lightest_mass

np.float64(2850.0)

<h3>INSIGHT: The result shows that the lighest female adelie has 2.8kg of body mass.</h3>

<h2>Question 2: What is the average body mass of male chinstrap from the year 2007? </h2

In [64]:
male_chinstrap_2007 = penguins[(penguins["species"] == "Chinstrap") & (penguins["sex"] == "male") & (penguins["year"] == 2007)]
average_mass = male_chinstrap_2007["body_mass_g"].mean()
male_chinstrap_2007

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
283  Chinstrap  Dream            51.3           18.2              197.0   
285  Chinstrap  Dream            51.3           19.9              198.0   
287  Chinstrap  Dream            51.7           20.3              194.0   
289  Chinstrap  Dream            52.0           18.1              201.0   
291  Chinstrap  Dream            50.5           19.6              201.0   
292  Chinstrap  Dream            50.3           20.0              197.0   
295  Chinstrap  Dream            49.2           18.2              195.0   
297  Chinstrap  Dream            48.5           17.5              191.0   
299  Chinstrap  Dream            50.6           19.4              193.0   
301  Chinstrap  Dream            52.0           19.0              197.0   

     body_mass_g   sex  year  mass_kg  
277       3900.0  male  2007      3.9  
278       3650.0  male  2007      3.6  
280       3725.0  male  2007      3.7  
283       3750.0  male  2007      3.8  
285       3700.0  male  2007      3.7  
287       3775.0  male  2007      3.8  
289       4050.0  male  2007      4.0  
291       4050.0  male  2007      4.0  
292       3300.0  male  2007      3.3  
295       4400.0  male  2007      4.4  
297       3400.0  male  2007      3.4  
299       3800.0  male  2007      3.8  
301       4150.0  male  2007      4.2

In [65]:
average_mass

np.float64(3819.230769230769)

<h3>Insight: This indicates the average body mass of ale chinstrap is 3.8 kg.</h3>

<h2>QUESTION 3 : Which island has the lightest penguin?</h2>

In [44]:
lightest_row = penguins[penguins["body_mass_g"] == penguins["body_mass_g"].min()]
island_lightest = lightest_row["island"].values[0]
lightest_row




species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
314  Chinstrap  Dream            46.9           16.6              192.0   

     body_mass_g     sex  year  mass_kg  
314       2700.0  female  2008      2.7

In [45]:
island_lightest

'Dream'

<h3>Insight: This indicates that Dream island has the lighest penguin weighing 2.7 kg.</h3>

<h2>Question 4: What is the difference between the heaviest and lightest penguin body mass in the year 2007?</h2

In [46]:
penguins_2007 = penguins[penguins["year"] == 2007]
max_mass = penguins_2007["body_mass_g"].max()
min_mass = penguins_2007["body_mass_g"].min()
mass_difference_2007 = max_mass - min_mass

print(f"Heaviest penguin in 2007: {max_mass} g")
print(f"Lightest penguin in 2007: {min_mass} g")
print(f"Difference: {mass_difference_2007} g")


Heaviest penguin in 2007: 6300.0 g
Lightest penguin in 2007: 2900.0 g
Difference: 3400.0 g


<h3>Insight: This indicates that the heaviest penguin in 2007 has 6.3kg while the lightest has 2.9 kg. This shows that their difference in body mass is 3.4 kg.</h3>

<h2>Question 5: From what specie has the heaviest body mass in the year 2007?</h2>

In [47]:
heaviest_2007 = penguins[penguins["year"] == 2007]["body_mass_g"].max()
species_heaviest_2007 = penguins[(penguins["year"] == 2007) & (penguins["body_mass_g"] == heaviest_2007)]["species"].values[0]
heaviest_2007


np.float64(6300.0)

In [48]:
species_heaviest_2007

'Gentoo'

<h3>Insight: This indicates that the heaviest penguin in 2007 is Gentoo with the body mass of 6.3 kg .</h3>

<h2>Question 6: What specie has the longest bill length in the year 2009?</h2>

In [49]:
penguins_2009 = penguins[penguins["year"] == 2009]
longest_bill = penguins_2009[penguins_2009["bill_length_mm"] == penguins_2009["bill_length_mm"].max()]
penguins_2009


species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
100     Adelie  Biscoe            35.0           17.9              192.0   
101     Adelie  Biscoe            41.0           20.0              203.0   
102     Adelie  Biscoe            37.7           16.0              183.0   
103     Adelie  Biscoe            37.8           20.0              190.0   
104     Adelie  Biscoe            37.9           18.6              193.0   
..         ...     ...             ...            ...                ...   
339  Chinstrap   Dream            55.8           19.8              207.0   
340  Chinstrap   Dream            43.5           18.1              202.0   
341  Chinstrap   Dream            49.6           18.2              193.0   
342  Chinstrap   Dream            50.8           19.0              210.0   
343  Chinstrap   Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  mass_kg  
100       3725.0  female  2009      3.7  
101       4725.0    male  2009      4.7  
102       3075.0  female  2009      3.1  
103       4250.0    male  2009      4.2  
104       2925.0  female  2009      2.9  
..           ...     ...   ...      ...  
339       4000.0    male  2009      4.0  
340       3400.0  female  2009      3.4  
341       3775.0    male  2009      3.8  
342       4100.0    male  2009      4.1  
343       3775.0  female  2009      3.8  

[120 rows x 9 columns]

In [50]:
longest_bill


species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
253  Gentoo  Biscoe            55.9           17.0              228.0   

     body_mass_g   sex  year  mass_kg  
253       5600.0  male  2009      5.6

In [51]:
longest_bill["species"].values[0]

'Gentoo'

<h3>Insight: This shows that specie that has the longest bill length in 2009 is Gentoo having 55.9 mm of length.</h3>

<h2>Question 7: From which sex of adelie has the shortest bill depth?</h2>

In [52]:
adelie = penguins[penguins["species"] == "Adelie"]
shortest_bill = adelie[adelie["bill_depth_mm"] == adelie["bill_depth_mm"].min()]
shortest_bill


species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
142  Adelie  Dream            32.1           15.5              188.0   

     body_mass_g     sex  year  mass_kg  
142       3050.0  female  2009      3.0

In [53]:
shortest_bill["sex"].values[0]

'female'

<h3>Insight: This shows that the female Adelie has the shortest bill depth having 15.5 depth.</h3>